opentelemetry-sdk/src/logs/mod.rs

```diff
@@ -57,4 +58,51 @@ mod tests {
             .expect("Attributes are expected");
         assert_eq!(attributes.len(), 2);
     }
+
+    #[test]
+    fn logging_sdk_shutdown_test() {
+        // Arrange
+        let exporter: InMemoryLogsExporter = InMemoryLogsExporter::default();
+        let logger_provider = LoggerProvider::builder()
+            .with_log_processor(SimpleLogProcessor::new(Box::new(exporter.clone())))
+            .build();
+        set_logger_provider(logger_provider);
+
+        // Act
+        let logger = logger("test-logger");
```

Comment:
this *MUST* occur after we set global::set_logger_provider, else it'll get permanently tied to noop provider. The current implementation does not have a mechanism to update already obtained logger with updated meter_provider. (same for traces,metrics too)

Just an observation. I'll send some docs updates to make this clear to end users. (open to discuss more on if this is the desired behavior or not!)